### ISSUE TITLE
[Q] Missing property in PaymentTerms?

### ISSUE THREAD

**Issue by sude22** [stephanstapel/ZUGFeRD-csharp#522]
Hello,
I have imported a Zugferd invoice, which has the field “ActualDiscountAmount” in the “ApplicableTradePaymentDiscountTerms”. This field is not present in PaymentTerms.cs. 
Is this a deliberate decision or has it simply been forgotten?

```
<?xml version="1.0" encoding="utf-8"?>
<rsm:CrossIndustryDocument xmlns:rsm="urn:ferd:CrossIndustryDocument:invoice:1p0"
                           xmlns:ram="urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:12"
                           xmlns:udt="urn:un:unece:uncefact:data:standard:UnqualifiedDataType:15">
	<rsm:SpecifiedExchangedDocumentContext>
...
			<ram:SpecifiedTradePaymentTerms>
				<ram:Description>Bis zum 24.01.2024 erhalten Sie 4,000  % Skonto, 2.74 EUR</ram:Description>
				<ram:DueDateDateTime>
					<udt:DateTimeString format="102">20240124</udt:DateTimeString>
				</ram:DueDateDateTime>
				<ram:ApplicableTradePaymentDiscountTerms>
					<ram:BasisAmount currencyID="EUR">68.5900</ram:BasisAmount>
					<ram:CalculationPercent>4.00</ram:CalculationPercent>
----->				<ram:ActualDiscountAmount currencyID="EUR">2.74</ram:ActualDiscountAmount>
				</ram:ApplicableTradePaymentDiscountTerms>
			</ram:SpecifiedTradePaymentTerms>
```

**Comment by stephanstapel** [comment]
Hi @sude22 ,
nobody requested it so far. If you like, please add it. It was available since ZUGFeRD 1.0. Can you also please add a positive test (value included) and negative test (value not included)?

**Comment by sude22** [comment]
Hi Stephan,
I mainly use your code to read in invoices and have not yet dealt with the finer details and tests.
I will take a closer look at it.

Thanks for your quick reply.

**Comment by KevinKrueger** [comment]
ApplicableTradePaymentDiscountTerms are only available in the Extended Profile.

**Comment by stephanstapel** [comment]
Yep, I was working yesterday on the SpecifiedTradePaymentTerms in general. Structure is much clearer now (I hope). Find the relevant part where writing this information could be added here:

https://github.com/stephanstapel/ZUGFeRD-csharp/blob/835ec2b4f93a48e0200bcbdc9b35d85bdff5af09/ZUGFeRD/InvoiceDescriptor23CIIWriter.cs#L1073

**Comment by KevinKrueger** [comment]
@stephanstapel Be careful not to test yourself to death.

![image](https://github.com/user-attachments/assets/21a78e63-eea8-471b-a49a-34a9a5d8e167)

`
if (paymentTerms.PaymentTermsType.HasValue && (paymentTerms.PaymentTermsType == PaymentTermsType.Skonto || paymentTerms.PaymentTermsType == PaymentTermsType.Verzug))
`

**Comment by stephanstapel** [comment]
> @stephanstapel Be careful not to test yourself to death.
> 
> ![image](https://private-user-images.githubusercontent.com/25382167/401540339-21a78e63-eea8-471b-a49a-34a9a5d8e167.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY0Mjc3MTcsIm5iZiI6MTczNjQyNzQxNywicGF0aCI6Ii8yNTM4MjE2Ny80MDE1NDAzMzktMjFhNzhlNjMtZWVhOC00NzFiLWE0OWEtMzRhOWE1ZDhlMTY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA5VDEyNTY1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliMDgwNDQ0NjJjYTVkMTg0NWYxNTY3OGJhZWM1ZWYxNDNlZWQ2ZjQyMmRkZWE5MjJjYWQ2ZTVlNDU2MTAxNjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5S2GXvnCnQQdieqsSl1TwCW_I8Tr79ftqhKFoe3OMos)
> 
> `if (paymentTerms.PaymentTermsType.HasValue && (paymentTerms.PaymentTermsType == PaymentTermsType.Skonto || paymentTerms.PaymentTermsType == PaymentTermsType.Verzug))`



not quite. Tag names are different!

**Comment by KevinKrueger** [comment]
@stephanstapel Yes, well, then just do the case substitution for `Writer.WriteStartElement(“ram”, ....);`

**Comment by stephanstapel** [comment]
amount property added
